src/java/org/apache/cassandra/index/sai/disk/v1/IndexWriterConfig.java

```diff
@@ -328,6 +361,32 @@ else if (options.get(MAXIMUM_NODE_CONNECTIONS) != null ||
                 similarityFunction = sourceModel.defaultSimilarityFunction;
             }
 
+            if (options.containsKey(VECTOR_FLUSH_THRESHOLD))
+            {
+                try
+                {
+                    vectorFlushThreshold = Integer.parseInt(options.get(VECTOR_FLUSH_THRESHOLD));
+                }
+                catch (IllegalArgumentException e)
+                {
+                    throw new InvalidRequestException(String.format("Vector flush threshold %s is not a valid integer for index %s",
+                                                                    options.get(VECTOR_FLUSH_THRESHOLD), indexName));
+                }
+            }
+
+            if (options.containsKey(VECTOR_FLUSH_PERIOD_IN_MS))
+            {
+                try
+                {
+                    vectorFlushPeriodInMs = Integer.parseInt(options.get(VECTOR_FLUSH_PERIOD_IN_MS));
```

Comment:
Similarly, this value must not be too frequent. It might even make sense to use seconds or minutes as the unit.

Reply:
now it's not user-facing, we don't need guardrails